lib/compat.php

```diff
@@ -568,3 +568,20 @@ function gutenberg_output_html_nav_menu_item( $item_output, $item, $depth, $args
 	return $item_output;
 }
 add_filter( 'walker_nav_menu_start_el', 'gutenberg_output_html_nav_menu_item', 10, 4 );
+
+/**
+ * Amends the paths to preload when initializing edit post.
+ *
+ * @see https://core.trac.wordpress.org/ticket/50606
+ *
+ * @since 8.4.0
+ *
+ * @param  array $preload_paths Default path list that will be preloaded.
+ * @return array Modified path list to preload.
+ */
+function gutenberg_preload_edit_post( $preload_paths ) {
+	$additional_paths = array( '/?context=edit' );
```

Comment:
This ticket has been rejected on Core, should we remove that code from here?

Reply:
Thanks for the heads up Riad 👍 This is on my radar -- will respond when I get the chance tomorrow.

Reply:
It's been quite a while since I've looked at this code, but yeah removing it makes sense to me, especially if, according to Timothy, the index endpoint doesn't even support `context` query params.

I'll draft something early next week.

Reply:
Here's an attempt at making the entity request urls more specific to each entity, and prevents the need for preloading `/?context=edit`: https://github.com/WordPress/gutenberg/pull/30482